python/mxnet/numpy/linalg.py

```diff
@@ -1532,3 +1569,104 @@ def eigh(a, upper=False):
     else:
         UPLO = 'U'
     return _mx_nd_np.linalg.eigh(a, UPLO)
+
+
+@wrap_np_binary_func
+def matmul(a: ndarray, b: ndarray, /, *, out: Optional[ndarray] = None, **kwargs) -> ndarray:
+    r"""Matrix product of two arrays.
+
+    Parameters
+    ----------
+    a, b : ndarray
+        Input arrays, scalars not allowed.
+    out : ndarray, optional
+        A location into which the result is stored.
+        If provided, it must have a shape that matches the signature (n,k),(k,m)->(n,m).
+        If not provided or None, a freshly-allocated array is returned.
+
+    Returns
+    -------
+    y : ndarray
+        The matrix product of the inputs.
+        This is a scalar only when both x1, x2 are 1-d vectors.
+
+    Raises
+    ------
+    MXNetError
+        If the last dimension of a is not the same size as the second-to-last dimension of b.
+        If a scalar value is passed in.
+
+    See Also
+    --------
+    tensordot : Sum products over arbitrary axes.
+    dot : alternative matrix product with different broadcasting rules.
+    einsum : Einstein summation convention.
+
+    .. note::
+
```

Comment:
Could you add note here as matmul in linalg is required by array-api-standard instead of official numpy and add matmul into __all__ (line 31 - 33)